test/net/NetTest.ooc

```diff
@@ -8,16 +8,75 @@
 
 use unit
 use net
+use concurrent
 
-//TODO: Write actual tests for source/net
-// Right now, this is just to make sure that the code is processed and compiled when testing
 NetTest: class extends Fixture {
 	init: func {
 		super("Net")
-		this add("No test", func {
-			expect(true)
+		this add("TCP Socket Client Server", func {
+			clientThread := Thread new( func {
+				ipText := t"0.0.0.0"
+				ip := IP4Address new(ipText toString())
+				i := 0
+				socket1 := TCPSocket new(SocketAddress new(ip, 8000))
+				socket1 connect()
+				c: Int
+				socket1 out write(i as Char)
+				while (i < 10) {
+					c = socket1 in read() as Int
+					expect(c == i)
+					if(c == i) {
+						i = i + 1
+						socket1 out write(i as Char)
+					}
+				}
+			})
+			ipText := t"0.0.0.0"
+			socket := ServerSocket new(ipText toString(), 8000)
+			socket listen()
+			clientThread start()
+			conn := socket accept()
+			c: Int
+			i := 0
+			while (i < 10) {
+				c = conn in read() as Int
+				expect(c == i)
+				conn out write(i as Char)
+				i = i + 1
+			}
+			expect(c == 9)
+			clientThread free()
+			ipText free()
+		})
+		this add("UDP Socket Client Readlines", func {
+			clientThread := Thread new( func {
+				ipText := t"0.0.0.0"
+				ip := IP4Address new(ipText toString())
+				socket := UDPSocket new(SocketAddress new(ip, 5000))
+				i := 0
+				correct := t"udp is fun"
+				while (i < 5) {
+					socket send(correct toString())
+					i = i + 1
+				}
+				correct free()
+			})
+			ipText := t"0.0.0.0"
+			ip := IP4Address new(ipText toString())
+			socket := UDPSocket new(SocketAddress new(ip, 5000))
+			socket bind()
+			clientThread start()
+			i := 0
+			expected := t"udp is fun"
+			while (i < 5) {
+				buffer := socket receive(128)
+				expect(buffer toString(), is equal to(expected toString()))
+				i = i + 1
+			}
+			clientThread free()
+			expected free()
+			ipText free()
```

Comment:
This call will propagate down to `OwnedBuffer free()` which will check `isOwned` and find it is `false` because owner is `Owner Static` and thus do nothing. `ipText` is a text literal and doesn't need to be freed.